core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -65,8 +93,12 @@ public boolean isConnectedToWifi() {
      * @return boolean true if a mobile network connection is available.
      */
     public boolean isConnectedToMobileNetwork() {
-        NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
-        return networkInfo != null && networkInfo.isConnected();
+        if (androidVersion.isMarshmallowOrHigher()) {
+            return connectedToNetworkTypeForMarshmallow(ConnectivityManager.TYPE_MOBILE);
+        } else {
+            NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
```

Comment:
These 2 lines are duplicated twice. Probably good idea to extract a little utility method that just receives the network type.